--- a/main.py
+++ b/main.py
@@ -118,7 +118,8 @@
 
     def on_configure_tray(tray_icon, item):
         logger.info("Configure from tray icon")
-        subprocess.Popen(f'"{MAIN_DIRECTORY}{glob.glob("configure.*", root_dir=MAIN_DIRECTORY)[0]}"', shell=True)
+        configure_script = glob.glob("configure.*", root_dir=MAIN_DIRECTORY)[0]
+        subprocess.Popen([sys.executable, f"{MAIN_DIRECTORY}{configure_script}"])
         clean_stop(tray_icon)
 
 
--- a/configure.py
+++ b/configure.py
@@ -475,16 +475,16 @@
             subprocess.Popen(["xdg-open", path])
 
     def on_theme_editor_click(self):
-        subprocess.Popen(
-            f'"{MAIN_DIRECTORY}{glob.glob("theme-editor.*", root_dir=MAIN_DIRECTORY)[0]}" "{self.theme_cb.get()}"',
-            shell=True)
+        theme_editor_script = glob.glob("theme-editor.*", root_dir=MAIN_DIRECTORY)[0]
+        subprocess.Popen([sys.executable, f"{MAIN_DIRECTORY}{theme_editor_script}", self.theme_cb.get()])
 
     def on_save_click(self):
         self.save_config_values()
 
     def on_saverun_click(self):
         self.save_config_values()
-        subprocess.Popen(f'"{MAIN_DIRECTORY}{glob.glob("main.*", root_dir=MAIN_DIRECTORY)[0]}"', shell=True)
+        main_script = glob.glob("main.*", root_dir=MAIN_DIRECTORY)[0]
+        subprocess.Popen([sys.executable, f"{MAIN_DIRECTORY}{main_script}"])
         self.window.destroy()
 
     def on_brightness_change(self, e=None):
